Auth와 Dialog 컴포넌트 + auth 관련하여 RTK 로직 추가, ErrorBoundary 컴포넌트 작성 및 Sentry 연동

hanana1253 opened the pull request:
## PR Type

What kind of change does this PR introduce?

<!-- 'x'를 이용하여 이 PR에 적용되는 항목을 확인해 주세요. -->

- [x] 버그를 수정했어요.
- [x] 새로운 기능을 추가했어요.
- [ ] 코드 스타일 업데이트를 했어요(포맷팅, 지역변수)
- [ ] 리팩토링을 했어요 (기능적인 변화 없이, api 변경 없이)
- [x] 환경설정을 변경했어요.
- [ ] 문서 내용을 변경했어요.
- [ ] 기타 사항을 설명해 주세요.

## Related Issues

closed #17 
closed #55 
solved #25 
solved #27 
#53  
closed #33 

<!--#을 눌러 이슈와 연결해 주세요-->

## What does this PR do?

<!--무엇을 하셨나요?-->

- [x] Auth 컴포넌트에 제목을 달고, 서버 인증 에러 메시지 컴포넌트 추가
- [x] Auth 스타일링 완성
- [x] Dialog 컴포넌트 스타일링 완성 
- [x] 인증정보 전역상태 rtk 로직 작성
- [x] 인증정보 관련하여 Header, Menu 컴포넌트에 구독에 따른 상태 변경 및 dispatch 로직 추가
- [x] Auth 컴포넌트 submit에 dispatch 로직 추가
- [x] ErrorBoundary 컴포넌트 작성 및 적용
- [x] Sentry for NextJS 적용    

## Other information

<!--참고한 자료, 추가적인 사항, 기타 의견-->

hanana1253 committed:
conf: 스토리북 테스트를 위해 템플릿에 #__next div 요소 추가

- 스토리북은 #root 로만 렌더링을 하고 있네요!
- nextJS가 사용하는 루트 요소인 __next 아이디를 갖는 div 요소를 추가했습니다.

hanana1253 committed:
refactor: dom 유틸함수에 브라우저환경을 위한 if 문 추가

브라우저 환경일 때 true를 반환하는 process.browser도 있지만 deprecated 될거라고 한다.
window 전역객체의 유무를 통해 확인한 후 dom 제어 코드를 실행하도록 조건문 처리해주자.

hanana1253 committed:
Merge branch 'develop' of https://github.com/TeamCooks/TwoSpoon into develop
t # Please enter a commit message to explain why this merge is necessary,

hanana1253 committed:
feat: AuthError 메시지 추가, 스타일링 완성

- 서버 인증 실패 시 뜨는 에러메시지 추가, 스타일링 완성

hanana1253 committed:
chore: Dialog 컴포넌트 스타일링 완성

- dialog 컴포넌트의 스타일링을 완성했습니다.

hanana1253 committed:
Merge branch 'feature/dialog' into develop

hanana1253 committed:
chore: Auth 컴포넌트에 as prop 추가

hanana1253 committed:
fix: CSS 문자열 오타 수정

hanana1253 committed:
fix: dom 유틸함수에 if 문 수정

- window type이 undefined가아닐 때 동작하게 했어야 했는데 반대로 해서 수정

hanana1253 committed:
fix: Search 요청 로직 수정

- params 로 전달할 경우 제대로 요청이 가지 않아서 url에 넣는 방식으로 수정

hanana1253 committed:
fix: Header 영역 margin-top 등 스타일링 추가

- header와 내용 요소 겹치지 않도록 추가했습니다

hanana1253 committed:
chore: headerHeight를 export로 내보내기

- Layout에서 사용하기 위해 headerHeight를 내보냈습니다.

hanana1253 committed:
feat: Auth 컴포넌트에 dispatch 로직 추가

hanana1253 committed:
feat: AuthSlice 작성 및 redux 로직 추가

hanana1253 committed:
chore: Auth 컴포넌트 key prop 추가& 테스트 스토리 파일 decorator 삭제

- key prop 추가
- .storybook/preview.js 에서 StoreProvider decorator 추가하면서 각 스토리파일에서는 삭제

hanana1253 committed:
conf: StoreProvider를 스토리북 환경에 추가

- redux store 구독 테스트를 위해 스토리북 환경에 StoreProvider wrapping

hanana1253 committed:
feat: Store에서 구독한 Auth 상태에 따른 menu / login 버튼 적용

- Header 우측 menu 또는 login 버튼에 적용될 store의 auth 상태 적용
- hide를 styled component에 맞게 프롭으로 추가

hanana1253 committed:
feat: signout 클릭 시 firebase & store auth 상태 변경 로직 추가

- signout 버튼 동작합니다.

hanana1253 committed:
chore: 변수명 변경 (data => results) 및 타입 정의

- 더 명확한 코드를 위해 변수명을 변경하고 타입을 정의했습니다.

hanana1253 committed:
Merge branch 'feature/search' into develop

hanana1253 committed:
fix: CSS 속성 오타 수정

- justify-content 에 제가 자꾸 s를 붙이네요...

hanana1253 committed:
chore: Transient prop으로 타입 정의 및 prop 이름 변경

- Header 컴포넌트에 hide 여부를 클래스가 아닌 prop으로 전달해서 styled 컴포넌트에 추가
- Auth 컴포넌트에서 타입 이름을 StyleInputProps에서 StyledInputProps로 변경

hustle-dev commented on src/store/slices/auth.ts:
```diff
@@ -0,0 +1,18 @@
+    signOut: (state: AuthState) => ({ authUser: null, isLoading: false }),
+  },
+});
+
```
AuthSlice에 대한 ReturnType을 사용해서 나중에 useSelector를 사용하는 부분에서 state의 타입을 정의해주면 될것 같아요

hanana1253 replied:
https://typeofnan.dev/how-to-use-the-useselector-redux-hook-with-typescript/

여기 참고해서 정의했어요 감사합니당~~ㅎㅎ

hhhyyo commented on src/components/Header/Header.styled.tsx:
```diff
@@ -1,21 +1,21 @@
   transition: top 0.2s ease-in-out;
   z-index: 10;
-
   &.hide {
```
이 부분 지우시면 될 것 같아요

hanana1253 replied:
네 지우고 푸시할게요!!

hustle-dev commented on src/pages/search/[keyword].tsx:
```diff
@@ -17,31 +19,31 @@ const Search: NextPage = ({ data }) => {
-  }).then((res) => res.json());
+export async function getServerSideProps({ query }: ContextProp) {
+  const { keyword } = query;
+  const { results, totalResults } = await fetch(
```
getServerSideProps와 같은 next.js의 환경에서 rtk query를 사용할 순 없을까요? 이 부분 알아보겠습니다.

hanana1253 replied:
저도 활용해보려 헀는데 서버사이드라서 RTK query가 제공하는 훅을 사용할 수 없다고만 나왔었어요 알아보고 저도 알려주세요!

hhhyyo reviewed (commented):
확인했습니다~

hanana1253 committed:
conf: Sentry 연동을 위한 패키지 설치 및 configuration

- @sentry/nextjs 설치하고 @sentry/wizard로 configuration 했습니다.
- next.config.js에 Sentry 관련 로직이 추가되었습니다.

hanana1253 committed:
feat: ErrorBoundary 적용 및 에러 페이지 커스텀, 에러 테스트 페이지 제작

- 에러 바운더리 컴포넌트를 적용하고, 에러페이지(_error.tsx)와 에러페이지 (/errorTest) 제작

hanana1253 committed:
feat: ErrorBoundary 컴포넌트 및 테스트 코드 작성

- 에러바운더리 컴포넌트를 작성했습니다.

hanana1253 committed:
conf: Sentry 패키지 설치로 yarn.lock 파일 변경

- 변경된 yarn.lock 파일입니다.

hanana1253 committed:
Merge branch 'feature/ErrorBoundary' into develop

hanana1253 committed:
chore: 필요없는 CSS 선택자 삭제

- 필요없는 코드 삭제했습니다.

hanana1253 committed:
chore: import 경로를 절대경로로 수정

- ErrorBoundary component/index에 추가
- _app.tsx에서 절대경로 import 구문으로 수정

hanana1253 committed:
fix: Signout 후 모달 창 꺼지지 않는 버그 해결

- 메뉴에서 signout 후 모달창이 꺼지지 않아서 authcontainer에 onclose 함수를 prop으로 전달해서 signup/signin 하면 showDialog 값을 변경하도록 했습니다.

hanana1253 committed:
fix: Auth Slice 기본값 수정 및 type 에러 해결

- RootState와 AuthState로 타입 명시하여 타입에러 해결
- 테스트로 두었던 authUser 기본값을 null로 수정

hanana1253 committed:
conf: @types/react-dom 패키지 설치

- react-dom에서 ESLint 문제가 나서 패키지 설치했습니다.

hanana1253 committed:
Merge branch 'develop' of https://github.com/TeamCooks/TwoSpoon into develop
